### PR TITLE
gh-105908: fix `barry_as_FLUFL` future import

### PR DESCRIPTION
**Pull request by thatbirdguythatuknownot**
In `compiler_setup()`, the `flags` parameter is a local struct that does not reflect changes to the pointer passed to `new_compiler()`. Fix by making the `flags` parameter a pointer to a struct instead of a struct.

* Issue: gh-105908

### CODE REVIEW

**Comment by iritkatriel** [comment]
Is there no test for this?

**Comment by thatbirdguythatuknownot** [comment]
> Is there no test for this?

I'm currently adding it.

File: Misc/NEWS.d/next/Core and Builtins/2023-06-19-11-04-01.gh-issue-105908.7oanny.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Fixed bug where gh-99111 breaks future import `barry_as_FLUFL` in the Python REPL.
```
Comment (Eclips4) [inline Misc/NEWS.d/next/Core and Builtins/2023-06-19-11-04-01.gh-issue-105908.7oanny.rst]:
```suggestion
Fixed bug where :gh:`99111` breaks future import ``barry_as_FLUFL`` in the Python REPL.
```

**Review by iritkatriel — APPROVED** [review]

**Comment by iritkatriel** [comment]
Thank you @thatbirdguythatuknownot .

**Comment by iritkatriel** [comment]
@thatbirdguythatuknownot there is a test failure in the "Address sanitizer" test. Is it due to this PR?

**Comment by thatbirdguythatuknownot** [comment]
> @thatbirdguythatuknownot there is a test failure in the "Address sanitizer" test. Is it due to this PR?

It must be from `flags = &_PyCompilerFlags_INIT`. I could probably bring back the `local_flags` variable to prevent it.

**Comment by sunmy2019** [comment]
Thanks! This change makes sense.

**Comment by JelleZijlstra** [comment]
Does this need to be backported to 3.12?

**Pull request merged**